app/models/user.rb

```diff
@@ -11,5 +11,6 @@ class User < ActiveRecord::Base
     has_secure_password
     has_many :items, dependent: :destroy
     has_many :reviews, dependent: :destroy
+    has_many :receipts, dependent: :destroy
```

Comment:
I agree it's too strong. Instead maybe have an after_destroy callback where you tweak the receipt in whatever way you want.